pkg/client/client_get.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+func (c *client) Get(ctx context.Context, prefix string, opOpts ...OpOption) (rev int64, err error) {
+	opt := opOption(opOpts)
+	if opt.response == nil {
+		return 0, fmt.Errorf("response is required")
+	}
+
+	prefix, single, err := opt.getPrefix(prefix)
+	if err != nil {
+		return 0, err
+	}
+
+	opts := []clientv3.OpOption{}
+
+	if single || opt.pageLimit == 0 {
+		if !single {
+			opts = append(opts, clientv3.WithPrefix())
+		}
+		resp, err := c.client.Get(ctx, prefix, opts...)
+		if err != nil {
+			return 0, err
+		}
+		for _, kv := range resp.Kvs {
+			r := &KeyValue{
+				Key:   kv.Key,
+				Value: kv.Value,
+			}
+			err = opt.response(r)
+			if err != nil {
+				return 0, err
+			}
+		}
+		return resp.Header.Revision, nil
+	}
+
+	opts = append(opts, clientv3.WithLimit(opt.pageLimit))
+	if opt.revision != 0 {
+		rev = opt.revision
+		opts = append(opts, clientv3.WithRev(rev))
```

Comment:
https://github.com/etcd-io/auger/blob/6d4429e755661e17dcbdb998c92bcf691230c7d1/pkg/client/client_get.go#L86-L89


This is to avoid adding the Rev option again